cclib/method/volume.py

```diff
@@ -224,11 +253,11 @@ def electrondensity(coords, mocoeffslist, gbasis, volume):
                     for j, yval in enumerate(y):
                         tmp = []
                         for zval in z:
-                            tmp.append(bfs[bs].amp(xval, yval, zval))
-                        data[i, j, :] = tmp
-                data *= mocoeff[bs]
+                            tmp.append(bfs[bs].amp(xval, yval, zval)) # first get amplitudes
+                        data[i, j, :] = tmp # this contains the amplitudes
+                data *= mocoeff[bs] # then multiply with mocoeffs until homos
                 wavefn += data
-            density.data += wavefn ** 2
+            density.data += wavefn ** 2 # lastly put them into second power
 
     # TODO ROHF

```

Comment:
I'm not too sure why this comment was left in the original implementation -- `electrondensity` function calculates the charge density from a user-designated list of orbitals anyway (which could be entire occupied orbitals or just a few frontier orbitals). If the calculation is restricted, it would just be fine to multiply by two here.

Reply:
Wouldn't that only be the case if it was a restricted closed shell system? For ROHF, we would need to only multiply the `wavefn**2` by 2 for the doubly occupied orbitals and by 1 otherwise?

Reply:
Yes -- but I thought that what the original author meant was to call the function using something like this for ROHF:
`volume.electrondensity(data.atomcoords[0], [data.mocoeffs[0][:data.homos[0]], data.mocoeffs[1][:data.homos[1]]
`
to indicate different number of orbitals to consider for ROHF calculations.

This is how the function was used in the test suite:
```
frontierorbs = [data_sp.mocoeffs[0][(data_sp.homos[0] - 3):(data_sp.homos[0] + 1)]]
        density = volume.electrondensity(data_sp.atomcoords[0],
                                         frontierorbs, data_basis.gbasis, vol)
```

Checking that inside the function makes a lot sense though and adding it would be pretty simple.

Reply:
We concluded that we aren't going to worry about this right now, right?

Reply:
@berquist Yes, that's correct.